dynamodb-stream-reader/src/main/java/elasticsearch/ElasticSearchRestClient.java

```diff
@@ -0,0 +1,112 @@
+        client = httpClient;
+        elasticSearchEndpointAddress = environment.readEnv(ELASTICSEARCH_ENDPOINT_ADDRESS_KEY);
+        elasticSearchEndpointIndex = environment.readEnv(ELASTICSEARCH_ENDPOINT_INDEX_KEY);
+        elasticSeachEndpointScheme = environment.readEnv(ELASTICSEARCH_ENDPOINT_API_SCHEME_KEY);
+        logger.info("using Elasticsearch endpoint {} {} and index {}",
+                elasticSeachEndpointScheme, elasticSearchEndpointAddress, elasticSearchEndpointIndex);
+    }
+
+    /**
+     * Adds or insert a document to an elasticsearch index.
+     * @param document the document to be inserted
+     * @return true if operation is an success
+     * @throws URISyntaxException thrown when uri is misconfigured
+     * @throws IOException thrown hen service i not available
+     * @throws InterruptedException thrown when service i interrupted
+     */
+    public boolean addDocumentToIndex(Map<String, String> document)
+            throws URISyntaxException, IOException, InterruptedException {
+        logger.debug("Upserting search index  with values {}", document);
+
+        String requestBody = "";
+        requestBody = JsonUtils.objectMapper.writeValueAsString(document);
+        String identifier = document.get("identifier");
+
+        HttpRequest request = HttpRequest.newBuilder()
+                .uri(createUpsertDocumentURI(identifier))
+                .header("Content-Type", "application/json")
+                .POST(HttpRequest.BodyPublishers.ofString(requestBody)) // GET is default
+                .build();
+
+
+        logger.debug("POSTing {} to endpoint {}", requestBody, createUpsertDocumentURI(identifier));
+        HttpResponse<String> response = doSend(request);
+        logger.debug(response.body());
+        return true;
+    }
+
+    /**
+     * Removes an indexed document from elasticsearch index.
+     * @param identifier identifier of document to remove from elasticsearch
+     * @return true if operatrion is successful
+     * @throws URISyntaxException thrown when uri is misconfigured
+     * @throws IOException thrown hen service i not available
+     * @throws InterruptedException thrown when service i interrupted
+     */
+    public boolean removeDocumentFromIndex(String identifier)
+            throws URISyntaxException, IOException, InterruptedException {
+        logger.trace("Deleting from search API publication with identifier: {}", identifier);
+
+        HttpRequest request = HttpRequest.newBuilder()
+                .uri(createUpsertDocumentURI(identifier))
+                .header("Content-Type", "application/json")
+                .DELETE()
+                .build();
+
+        logger.debug("DELETEing {} ", createUpsertDocumentURI(identifier));
+        HttpResponse<String> response = doSend(request);
+        logger.debug(response.body());
+
+        return true;
+    }
+
+    public HttpResponse<String> doSend(HttpRequest request) throws IOException, InterruptedException {
```

Comment:
This is public for usage in a test with no assertion. Re-evaluate if you need the test.

Reply:
IT's public to keep mockito happy when doing test code coverage stuff....

Reply:
make private after test refactoring